test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -0,0 +1,1002 @@
+							args: { key: 'url_custom_field' },
+						},
+					},
+				},
+			},
+		},
+	],
+};
+// Image blocks.
+const urlBindingImageBlock = {
+	name: 'core/image',
+	attributes: {
+		url: imagePlaceholder,
+		alt: 'a',
+		title: 't',
+		metadata: {
+			bindings: {
+				url: {
+					source: 'core/post-meta',
+					args: { key: 'url_custom_field' },
+				},
+			},
+		},
+	},
+};
+const altBindingImageBlock = {
+	name: 'core/image',
+	attributes: {
+		url: imagePlaceholder,
+		alt: 'a',
+		title: 't',
+		metadata: {
+			bindings: {
+				alt: {
+					source: 'core/post-meta',
+					args: { key: 'text_custom_field' },
+				},
+			},
+		},
+	},
+};
+const titleBindingImageBlock = {
+	name: 'core/image',
+	attributes: {
+		url: imagePlaceholder,
+		alt: 'a',
+		title: 't',
+		metadata: {
+			bindings: {
+				title: {
+					source: 'core/post-meta',
+					args: { key: 'text_custom_field' },
+				},
+			},
+		},
+	},
+};
+const multipleBindingsImageBlock = {
+	name: 'core/image',
+	attributes: {
+		url: imagePlaceholder,
+		alt: 'a',
+		title: 't',
+		metadata: {
+			bindings: {
+				url: {
+					source: 'core/post-meta',
+					args: { key: 'url_custom_field' },
+				},
+				alt: {
+					source: 'core/post-meta',
+					args: { key: 'text_custom_field' },
+				},
+			},
+		},
+	},
+};
+// Image variables.
+const imageReplaceName = 'Replace';
+const imageAltLabel = 'Alternative text';
+const imageTitleLabel = 'Title attribute';
+
+test.describe( 'Block bindings - Post/page context', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	// Paragraph block tests.
+	test( 'Paragraph - should show the value of the custom field when exists', async ( {
+		editor,
+	} ) => {
+		await editor.insertBlock( contentBindingParagraphBlock );
+		const paragraphBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+		const paragraphContent = await paragraphBlock.textContent();
+		expect( paragraphContent ).toBe( textCustomFieldValue );
+	} );
+
+	test( "Paragraph - should show the value of the key when custom field doesn't exists", async ( {
+		editor,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: {
+				content: 'p',
+				metadata: {
+					bindings: {
+						content: {
+							source: 'core/post-meta',
+							args: { key: 'non_existing_custom_field' },
+						},
+					},
+				},
+			},
+		} );
+		const paragraphBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+		const paragraphContent = await paragraphBlock.textContent();
+		expect( paragraphContent ).toBe( 'non_existing_custom_field' );
```

Comment:
Would it make sense to add additional check whether the RichText field is readonly?